api/classes/publisher.js

```diff
@@ -66,32 +50,42 @@ function getPublishedFiles(publishedProject) {
 /**
  * Publish helpers
  */
-function uploadFile(file, root) {
+function upload(path, buffer, mimeType, resolve, reject) {
+  var headers = {
+    'Cache-Control': 'private max-age=0',
+    'Content-Type': mimeType,
+    'Content-Length': buffer.length,
+    'x-amz-acl': 'public-read'
+  };
+
+  var request = client.put(path, headers);
+  request.on('error', function(err) {
+    reject(err);
+  });
+  request.on('continue', function() {
+    request.end(buffer);
+  });
+  request.on('response', function(res) {
+    if (res.statusCode === 200) {
+      log.info('Successfully uploaded ' + path);
+      resolve();
+    } else {
+      reject('S3 upload returned ' + res.statusCode);
+    }
+  });
+}
+
+function uploadFile(file, root, options) {
   return new Promise(function(resolve, reject) {
     var buffer = file.get('buffer');
     var path = root + file.get('path');
-    var headers = {
-      'Cache-Control': 'private max-age=0',
-      'Content-Type': mime.lookup(path),
-      'Content-Length': buffer.length,
-      'x-amz-acl': 'public-read'
-    };
+    var mimeType = mime.lookup(path);
 
-    var request = client.put(path, headers);
-    request.on('error', function(err) {
-      reject(err);
-    });
-    request.on('continue', function() {
-      request.end(buffer);
-    });
-    request.on('response', function(res) {
-      if (res.statusCode === 200) {
-        log.info('Successfully uploaded ' + path);
-        resolve();
-      } else {
-        reject('S3 upload returned ' + res.statusCode);
-      }
-    });
+    if(mimeType === 'text/html') {
+      buffer = new Buffer(Remix.inject(buffer.toString(), options));
+    }
+
+    upload(path, buffer, mimeType, resolve, reject);
```

Comment:
As a followup, if you really want to keep it then don't instantiate the promise here. Do it in `upload()` and just return:

```js
return upload(path, buffer, mimeType);
```